app/src/main/java/ch/epfl/sdp/drone3d/ui/auth/AuthActivity.kt

```diff
@@ -62,14 +61,22 @@ abstract class AuthActivity : AppCompatActivity()  {
                     // If sign in fails, display a message to the user.
                     ToastHandler.showToast(baseContext, failMessage)
                     if (task.exception?.message != null) {
-                        infoText.text = task.exception?.message
-                        infoText.setTextColor(Color.RED)
-                        infoText.visibility = View.VISIBLE
+                        writeErrorMessage(task.exception?.message!!)
```

Comment:
Do you really need the !! even though you check that it is not null in the if?

Reply:
Yes, the ide asked me to put it